### PR TITLE
UPSTREAM: <carry>: Upgrade DSP api/go.mod package versions

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**

resolves: https://issues.redhat.com/browse/RHOAIENG-7308

Implemented fixes on my fork and tested out by importing the fork in my personal snyk account and making sure the CVEs were eliminated.
**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between f3fbc2cfba32098ae1902ec8afa8d9661a2b30b0...2bf529fe091b2bcb589e7ed90990fec428ba6fd4
```

**Comment by dsp-developers** [comment]
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-54.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-54
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-54"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-54"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-54"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-54"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-54"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/mlmd-grpc-server:latest"
    envoy:
      image: "registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-54"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/54/head
git checkout -b pullrequest 2bf529fe091b2bcb589e7ed90990fec428ba6fd4
oc apply -f dspa.pr-54.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...a54c07ff358a53ea47962928d376657a195d24bf
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-54`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-54`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-54`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-54`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-54`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-54`

**Review by amadhusu — APPROVED** [review]

File: api/go.mod [hunk 1]
```diff
@@ -10,5 +10,6 @@ require (
 replace (
 	github.com/mattn/go-sqlite3 => github.com/mattn/go-sqlite3 v1.14.18
 	golang.org/x/net => golang.org/x/net v0.17.0
+	golang.org/x/net v0.17.0 => golang.org/x/net v0.23.0
```
Comment (gmfrasca) [inline api/go.mod:13]: does this not just reduce the above line down to 
`golang.org/x/net => golang.org/x/net v0.23.0`?

**Comment by gmfrasca** [comment]
since you updated `go.mod`, you'll need to update the license CSVs. you can do that by simply running `make all` in `backend/`

**Comment by DharmitD** [comment]
> since you updated `go.mod`, you'll need to update the license CSVs. you can do that by simply running `make all` in `backend/`

resolved this in https://github.com/opendatahub-io/data-science-pipelines/pull/56

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 50c0275e60ebd028c36b94dae17c0aacf0d03afd...a54c07ff358a53ea47962928d376657a195d24bf
```

**Pull request merged**